sandbox/apparmor/notify/ioctl.go

```diff
@@ -0,0 +1,142 @@
+package notify
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"log"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/snapcore/snapd/arch"
+	"github.com/snapcore/snapd/osutil"
+)
+
+var ErrIoctlReturnInvalid = errors.New("IOCTL request returned invalid bufsize")
+
+var doSyscall = func(trap, a1, a2, a3 uintptr) (r1, r2 uintptr, err unix.Errno) {
+	return unix.Syscall(trap, a1, a2, a3)
+}
+
+type hexBuf []byte
+
+func (hb hexBuf) String() string {
```

Comment:
this doesn't seem to be tested

Reply:
Thanks. I just added a test for this, as well as the buffer contents before and after being passed to ioctl(2).